medic/cht-core
# Extend add-local-ip-certs-to-docker.sh to install expired and self signed certs

# Description

In working on some debug stuff for [this PR](https://github.com/medic/cht-core/pull/7374), I really wanted an easy way to install self signed and expired certs.  I realized maybe some other people might too!

Better than some other [rando repo](https://github.com/mrjones-plip/mrjones-medic-scratch/blob/main/add-expired-local-ip-certs-to-docker.sh) ;)

Maybe not the right location for this after our efforts in #7413 - happy to move elsewhere if so!

## Testing

I had some issues getting this to work and found this simple test harness really helpful:
1. create a 1 line bash script `checkcert.sh`:
   ```bash
   echo|openssl s_client -servername 127.0.0.1 -connect 127.0.0.1:443 | openssl x509 -noout -dates
   ```
1. make it executable `chmod +x checkcert.sh`
2. run it every half second: `watch -n 0.5 ./checkcert.sh`

This way `openssl` will tell which cert is installed and when it expires.  

Another simple check (because I thought I was going crazy) which I'll add, is a simple way to check the hash of the `crt` and `key` files:
1. create a 1 line bash script `get-hash.sh`:
   ```bash
   docker exec -it medic-os bash -c "md5sum /srv/settings/medic-core/nginx/private/default.*"
   ```
1. make it executable `chmod +x get-hash.sh`
2. run it every half second: `watch -n 0.5 ./get-hash.sh`

The beauty of both solutions is that you can run `docker-compose down&&docker volume rm medic-data&&docker-compose up` and both `watch` commands will happily fail and then succeed across these very destructive docker commands.  `watch` for the win!

## Code review checklist

## License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.
